Another thing I noticed is that (also on OS X) `zpm list` returns a weird character:

     ▲ ~/Playground/zpm-zsh make test
    gcc -Wall -o zpm -g zpm.c
    cram ./zpm.t
    !
    --- zpm.t
    +++ zpm.t.err
    @@ -63,6 +63,7 @@
       [1]
       $ zpm list
       rupa/z .* (re)
    +  \xf0 (esc)

     Can install multiple plugins under the same user.


    # Ran 1 tests, 0 skipped, 1 failed.
    make: *** [test] Error 1

I believe this is a off-by-one error. Should check the implementation a bit further.

--- a/zpm.t
+++ b/zpm.t
@@ -115,14 +115,14 @@ Plugin directory is properly unlinked after remove
   $ zpm "zsh-users/zsh-syntax-highlighting" > /dev/null
   $ zpm remove "zsh-users/zsh-syntax-highlighting" > /dev/null
   $ ls ~/.zpm/plugins/zsh-users/zsh-syntax-highlighting
-  ls: cannot access '.*/.zpm/plugins/zsh-users/zsh-syntax-highlighting': No such file or directory (re)
+  ls: cannot access ?'(.*).*/.zpm/plugins/zsh-users/zsh-syntax-highlighting?'(.*): No such file or directory (re)
   [2]
 
 Parent directory is properly removed if necessary
   $ zpm "zsh-users/zsh-autosuggestions"         > /dev/null
   $ zpm remove "zsh-users/zsh-autosuggestions"  > /dev/null
   $ ls ~/.zpm/plugins/zsh-users
-  ls: cannot access '.*/.zpm/plugins/zsh-users': No such file or directory (re)
+  ls: cannot access ?'(.*).*/.zpm/plugins/zsh-users?'(.*): No such file or directory (re)
   [2]
 
 Plugin directory is not removed after disable

--- a/zpm.t
+++ b/zpm.t
@@ -115,14 +115,14 @@ Plugin directory is properly unlinked after remove
   $ zpm "zsh-users/zsh-syntax-highlighting" > /dev/null
   $ zpm remove "zsh-users/zsh-syntax-highlighting" > /dev/null
   $ ls ~/.zpm/plugins/zsh-users/zsh-syntax-highlighting
-  ls: cannot access ?'(.*).*/.zpm/plugins/zsh-users/zsh-syntax-highlighting?'(.*): No such file or directory (re)
+  ls: cannot access .*/.zpm/plugins/zsh-users/zsh-syntax-highlighting: No such file or directory (re)
   [2]
 
 Parent directory is properly removed if necessary
   $ zpm "zsh-users/zsh-autosuggestions"         > /dev/null
   $ zpm remove "zsh-users/zsh-autosuggestions"  > /dev/null
   $ ls ~/.zpm/plugins/zsh-users
-  ls: cannot access ?'(.*).*/.zpm/plugins/zsh-users?'(.*): No such file or directory (re)
+  ls: cannot access (.*).*/.zpm/plugins/zsh-users: No such file or directory (re)
   [2]
 
 Plugin directory is not removed after disable

